sunnypilot/selfdrive/controls/lib/lane_turn_desire.py

```diff
@@ -32,14 +43,32 @@ def update_params(self) -> None:
     self.param_read_counter += 1
 
   def update_lane_turn(self, blindspot_left: bool, blindspot_right: bool, left_blinker: bool, right_blinker: bool, v_ego: float) -> None:
-    if left_blinker and not right_blinker and v_ego < self.lane_turn_value and not blindspot_left:
+    if self.lane_turn_type != TURN_STATE['OFF'] and left_blinker and not right_blinker and v_ego < self.lane_turn_value and not blindspot_left:
       self.turn_direction = custom.TurnDirection.turnLeft
-    elif right_blinker and not left_blinker and v_ego < self.lane_turn_value and not blindspot_right:
+    elif self.lane_turn_type != TURN_STATE['OFF'] and right_blinker and not left_blinker and v_ego < self.lane_turn_value and not blindspot_right:
       self.turn_direction = custom.TurnDirection.turnRight
     else:
       self.turn_direction = custom.TurnDirection.none
 
-  def get_turn_direction(self):
-    if not self.enabled:
-      return custom.TurnDirection.none
+  def return_desire(self, steering_pressed: bool, steering_torque: float) -> log.Desire:
+    # disabled or no turn -> return none
+    if self.lane_turn_type == TURN_STATE['OFF'] or self.turn_direction == custom.TurnDirection.none:
+      return log.Desire.none
+
+    if self.lane_turn_type == TURN_STATE['NUDGE']:
+      # In nudge mode, require steering press for turn desire
+      turn_torque_applied = steering_pressed and ((steering_torque > 0 and self.turn_direction == custom.TurnDirection.turnLeft) or
+                             (steering_torque < 0 and self.turn_direction == custom.TurnDirection.turnRight))
+      if not turn_torque_applied:
+        return log.Desire.none
+
+    return TURN_DESIRES[self.turn_direction]
+
+  def update(self, blindspot_left: bool, blindspot_right: bool, left_blinker: bool, right_blinker: bool, v_ego: float,
+             steering_pressed: bool, steering_torque: float) -> log.Desire:
+    self.update_params()
+    self.update_lane_turn(blindspot_left, blindspot_right, left_blinker, right_blinker, v_ego)
+    return self.return_desire(steering_pressed, steering_torque)
```

Comment:
Instead of returning the method call directly, assign its result to a local variable (such as `desire = self.return_desire(...)`) before returning. It makes debugging easier and a clearer flow to follow.